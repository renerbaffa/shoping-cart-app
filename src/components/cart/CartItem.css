.CartItem-container {
  width: 100%;
  display: flex;
  border-top: 1px solid lightgray;
  border-left: 1px solid lightgray;
  border-right: 1px solid lightgray;
}

// need to work here
.ProductsList-container .CartItem-container:last-child {
  border-bottom: 1px solid lightgray;
}

.CartItem-container-odd {
  background-color: #f2f2f2;
}

.CartItem-image {
  height: 50px;
  width: calc(12% - 6px);
  margin: 3px;
}

.CartItem-description-container {
  padding: 2px 12px;
  width: 35%;
}

.CartItem-quantity {
  width: 33%;
  align-self: center;
  display: flex;
  justify-content: center;
}

.CartItem-price {
  width: 10%;
  align-self: center;
  display: flex;
  justify-content: center;
}

.CartItem-stock {
  width: 10%;
  align-self: center;
  display: flex;
  justify-content: center;
}

.CartItem-name {
  font-weight: bold;
  font-size: 14px;
}

.CartItem-description {
  font-size: 13px;
}
