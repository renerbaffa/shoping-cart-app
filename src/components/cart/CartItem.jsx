import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import withProduct from '../../HOCs/withProduct';

import Image from '../shared/Image';

import './CartItem.css';

export const CartItem = ({
  className,
  description,
  image,
  index,
  name,
  onAddProduct,
  quantity,
  unitPrice,
  unitsInStock,
  ...other
}) => (
  <div className={cx(
    className,
    'CartItem-container',
    { 'CartItem-container-odd' : index % 2 }
  )}>
    <Image
      className="CartItem-image"
      alt={name}
      src={image}
    />
    <div className="CartItem-name">{name}</div>
    <div className="CartItem-price">Price: <b>${unitPrice}</b></div>
    <div className="CartItem-quantity">Quantity: <b>{quantity}</b></div>
    <div className="CartItem-stock">
      <b style={{ paddingRight: 3 }}>{unitsInStock}</b> in stock
    </div>
    <div className="CartItem-total">
      <b>Total</b>
      <b>$ {quantity * unitPrice}</b>
    </div>
  </div>
);

CartItem.propTypes = {
  className: PropTypes.string,
  description: PropTypes.string,
  image: PropTypes.string,
  index: PropTypes.number,
  name: PropTypes.string,
  onAddProduct: PropTypes.func,
  quantity: PropTypes.number,
  unitPrice: PropTypes.number,
  unitsInStock: PropTypes.number,
}

CartItem.defaultProps = {
  className: '',
  description: '',
  image: '',
  index: 0,
  name: '',
  onAddProduct: () => {},
  quantity: 0,
  unitPrice: 0,
  unitsInStock: 0,
}

export default withProduct(CartItem);
