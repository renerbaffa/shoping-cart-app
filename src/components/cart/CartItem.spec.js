import React from 'react';
import renderer from 'react-test-renderer';

import { CartItem } from './CartItem';

describe('<CartItem />', () => {
  let component;

  beforeEach(() => {
    component = (
      <CartItem
        className="className"
        description="description"
        image="image"
        index={0}
        name="Product Name"
        onAddProduct={() => {}}
        quantity={3}
        unitPrice={3}
        unitPrice={29.30}
        unitsInStock={12}
      />
    );
  });

  it('should render correctly', () => {
    const cartItem = renderer.create(component).toJSON();
    expect(cartItem).toMatchSnapshot();
  });

  it('should reach 100% of coverage', () => {
    CartItem.defaultProps.onAddProduct();
    CartItem.defaultProps.onRemoveProduct();
  });
});
