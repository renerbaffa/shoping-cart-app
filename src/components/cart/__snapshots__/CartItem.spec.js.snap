// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should render correctly 1`] = `
<div
  className="className CartItem-container"
>
  <img
    alt="Product Name"
    className="CartItem-image"
    src="image"
  />
  <div
    className="CartItem-name"
  >
    Product Name
  </div>
  <div
    className="CartItem-price"
  >
    Price: 
    <b>
      $
      29.3
    </b>
  </div>
  <div
    className="CartItem-quantity"
  >
    Quantity: 
    <b>
      3
    </b>
  </div>
  <div
    className="CartItem-stock"
  >
    <b
      style={
        Object {
          "paddingRight": 3,
        }
      }
    >
      12
    </b>
     in stock
  </div>
  <div
    className="CartItem-total"
  >
    <b>
      Total
    </b>
    <b>
      $ 
      87.9
    </b>
  </div>
</div>
`;
